pkg/deploy/broker.go

```diff
@@ -0,0 +1,200 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package deploy
+
+import (
+	"fmt"
+
+	"github.com/submariner-io/submariner-operator/internal"
+	"github.com/submariner-io/submariner-operator/internal/image"
+	"github.com/submariner-io/submariner-operator/pkg/broker"
+
+	"github.com/submariner-io/admiral/pkg/stringset"
+	"github.com/submariner-io/submariner-operator/pkg/discovery/globalnet"
+	"github.com/submariner-io/submariner-operator/pkg/subctl/cmd/utils/restconfig"
+	"github.com/submariner-io/submariner-operator/pkg/subctl/components"
+	v1 "k8s.io/api/core/v1"
+
+	submarinerv1a1 "github.com/submariner-io/submariner-operator/api/submariner/v1alpha1"
+	"github.com/submariner-io/submariner-operator/pkg/internal/cli"
+	"github.com/submariner-io/submariner-operator/pkg/subctl/datafile"
+	"github.com/submariner-io/submariner-operator/pkg/subctl/operator/brokercr"
+	"github.com/submariner-io/submariner-operator/pkg/subctl/operator/submarinerop"
+)
+
+type DeployOptions struct {
+	OperatorDebug               bool
+	GlobalnetEnable             bool
+	IpsecSubmFile               string
+	GlobalnetCIDRRange          string
+	DefaultGlobalnetClusterSize uint
+	ComponentArr                []string
+	GlobalCIDRConfigMap         *v1.ConfigMap
+	DefaultCustomDomains        []string
+	Repository                  string
+	ImageVersion                string
+	BrokerNamespace             string
+}
+
+var ValidComponents = []string{components.ServiceDiscovery, components.Connectivity}
+
+const brokerDetailsFilename = "broker-info.subm"
+
+func Broker(do DeployOptions, kubeConfig, kubeContext string) error {
+	status := cli.NewStatus()
```

Comment:
This doesn’t work in a library context. The status handler must be provided to the function, as done in `cloud-prepare`.

Reply:
Yes. I will make those changes as part of another PR.